client/src/holdings/AddTradeDialog.tsx

```diff
@@ -48,101 +122,111 @@ export function AddTradeDialog({
-              paymentCurrency: 'GBP', // Assume trade was always paid for in GBP for now.
-              fee: Number(values.fee ?? 0),
-              tax: Number(values.tax ?? 0),
-              fxRate,
-            };
-            try {
-              const trade = await API.createTrade(input);
-              onTradeCreated(trade);
-              setOpen(false);
-            } catch (err) {
-              showNotification(
-                `Create trade for ${holding.symbol} failed!`,
-                'error'
-              );
-            }
-            setSubmitting(false);
-          }}>
-          {({isSubmitting}) => (
-            <Form>
-              <DialogTitle id='form-dialog-title'>
-                Add Trade ({holding.symbol})
-              </DialogTitle>
-              <DialogContent className='DialogContent'>
-                <MuiPickersUtilsProvider utils={DateFnsUtils}>
-                  <Field
-                    component={KeyboardDateTimePicker}
-                    label='Date'
-                    variant='inline'
-                    name='date'
-                  />
-                </MuiPickersUtilsProvider>
-                <Field as='select' name='category'>
-                  <option value='BUY'>Buy</option>
-                  <option value='SELL'>Sell</option>
-                </Field>
-                <Field component={TextField} label='Broker' name='broker' />
-                <Field component={TextField} label='Quantity' name='quantity' />
-                <Field as='select' name='priceCurrency'>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <DialogTitle id='form-dialog-title'>
+            Add Trade ({holding.symbol})
+          </DialogTitle>
+          <DialogContent className='DialogContent'>
+            <div className='Category'>
+              <Controller
+                render={(category) => (
+                  <ToggleButtonGroup
+                    exclusive
+                    size={'medium'}
+                    aria-label='Category'
+                    {...category}
+                    onChange={(e, value) => {
+                      category.onChange(value);
+                    }}>
+                    <ToggleButton value='BUY' key='BUY'>
+                      BUY
+                    </ToggleButton>
+                    <ToggleButton value='SELL' key='SELL'>
+                      SELL
+                    </ToggleButton>
+                  </ToggleButtonGroup>
+                )}
+                label='Category'
+                name='category'
+                control={control}
+                inputRef={register}
+              />
+            </div>
+            <MuiPickersUtilsProvider utils={DateFnsUtils}>
+              <DateTimePicker
+                value={selectedDate}
+                onChange={handleDateChange}
+                label='Date'
+                variant='inline'
+                name='date'
+                inputRef={register}
+              />
```

Comment:
Your code wasn't actually working for dates. I found a way to do it using Controller:

```ts
<Controller
name='date'
label='Date'
control={control}
render={({onChange, value}) => (
  <DateTimePicker
    value={value}
    onChange={(date) => onChange(date)}
    variant='inline'
    inputRef={register}
  />
)}
/>
```